client/components/seo/seo-site-search-preview.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { PropTypes } from 'react';
+
+export const SeoSiteSearchPreview = React.createClass( {
+	shouldComponentUpdate( next ) {
```

Comment:
Any aversion to [`react-pure-render/mixin`](https://github.com/gaearon/react-pure-render#mixin)? A bit disappointing that it's now deprecated, but I don't see any reason it wouldn't work as we'd intend it to. Otherwise, we could pull in [`react-addons-shallow-compare`](https://facebook.github.io/react/docs/shallow-compare.html) and achieve the same here (it's planned to be added as a dependency in #5647 anyways).